spec.html

```diff
@@ -11643,33 +11643,58 @@ <h1>Realms</h1>
       </table>
     </emu-table>
 
-    <emu-clause id="sec-initializehostdefinedrealm" type="abstract operation" oldids="sec-createrealm,sec-setrealmglobalobject">
-      <h1>InitializeHostDefinedRealm ( ): either a normal completion containing ~unused~ or a throw completion</h1>
+    <emu-clause id="sec-makerealm" type="abstract operation" oldids="sec-initializehostdefinedrealm,sec-createrealm,sec-setrealmglobalobject">
+      <h1>
+        MakeRealm (
+          _customizations_: an Abstract Closure that takes a Realm Record and returns a List of 2 elements, each of which is either an Object or *undefined*,
```

Comment:
It might be slightly nicer for callers if they can omit the second return value (or return a single value or a list). Only window agents need this complexity, everything else has them be the same.

(Also, thanks for working on this, love to see this made more formal!)

Reply:
In the HTML spec, I believe that would change 2 lines of the form
```
Return « <var>global</var>, undefined ».
```
to
```
Return « <var>global</var> ».
```
or
```
Return <var>global</var>.
```
I suppose it's slightly nicer, but I'm doubtful that it's worth the increased complexity for IHDR.

Reply:
There's also Service Workers and probably everything else that wants to build on the JS specification that isn't HTML, but fair enough to not optimize for that as it's largely theoretical.